lib/model/model.go

```diff
@@ -1742,22 +1742,17 @@ func sendIndexTo(prevSequence int64, conn protocol.Connection, folder string, fs
 	var err error
 	var f protocol.FileInfo
 	batch := newFileInfoBatch(nil)
-	debugMsg := func(t string) string {
-		return fmt.Sprintf("Sending indexes for %s to %s at %s: %d files (<%d bytes) (%s)", folder, deviceID, conn, len(batch.infos), batch.size, t)
-	}
-	msg := [2]string{"initial index", "batched update"}
 	batch.flushFn = func(fs []protocol.FileInfo) error {
+		l.Debugf("Sending indexes for %s to %s at %s: %d files (<%d bytes)", folder, deviceID, conn, len(batch.infos), batch.size)
 		if initial {
-			if err = conn.Index(folder, fs); err != nil {
-				return err
-			}
-			l.Debugln(debugMsg(msg[0]))
 			initial = false
-		} else {
-			if err = conn.IndexUpdate(folder, fs); err != nil {
+			if err = conn.Index(folder, fs); err != nil {
 				return err
 			}
-			l.Debugln(debugMsg(msg[0]))
+			return nil
+		}
+		if err = conn.IndexUpdate(folder, fs); err != nil {
```

Comment:
actually, `return conn.Index…` should suffice for both of these

Reply:
or wait, `err` is shadowed? is that intentional? maybe I need to read this again

Reply:
It looks to me like the error is properly handled in the return path, so it should not be necessary to closure over it....

Reply:
The return thing was a stupid oversight and the error declaration is in place for `WithHaveSequence` later on - I moved that back to make that more clear.